test/unit_tests/braket/timings/test_time_series.py

```diff
@@ -54,6 +54,123 @@ def test_get_sorted(default_values, default_time_series):
+    assert constant_ts.values() == [3.14] * len(times)
+
+
+def test_periodic_signal():
+    times = list(range(4))
+    values = [0, 1, 3, 0]
+    new_ts = TimeSeries.periodic_signal(times=times, values=values, num_repeat=3)
+    expected_times = list(range(10))
+    expected_values = [0, 1, 3, 0, 1, 3, 0, 1, 3, 0]
+
+    assert new_ts.times() == expected_times
+    assert new_ts.values() == expected_values
+
+
+@pytest.mark.xfail(raises=ValueError)
+def test_periodic_signal_not_eq_length():
+    times = list(range(5))
+    values = [0.5, 1, 1, 0]
+    TimeSeries.periodic_signal(times=times, values=values, num_repeat=3)
+
+
+def test_concatenate():
+    times_1 = list(range(4))
+    values_1 = [0.5, 1, 1, 0]
+    time_series_1 = TimeSeries.from_lists(times=times_1, values=values_1)
+
+    times_2 = list(range(4, 8))
+    values_2 = [-0.5, -1, -1, 0]
+    time_series_2 = TimeSeries.from_lists(times=times_2, values=values_2)
+
+    new_ts = time_series_1.concatenate(time_series_2)
+
+    assert new_ts.times() == times_1 + times_2
+    assert new_ts.values() == values_1 + values_2
+
+    new_ts = time_series_1.concatenate(TimeSeries())
+    assert new_ts.times() == times_1
+    assert new_ts.values() == values_1
+
+
+@pytest.mark.xfail(raises=ValueError)
+def test_concatenate_not_ordered():
+    times_1 = list(range(4))
+    values_1 = [0.5, 1, 1, 0]
+    time_series_1 = TimeSeries.from_lists(times=times_1, values=values_1)
+
+    times_2 = list(range(4))
+    values_2 = [-0.5, -1, -1, 0]
+    time_series_2 = TimeSeries.from_lists(times=times_2, values=values_2)
+
+    time_series_1.concatenate(time_series_2)
+
+
+def test_from_lists():
+    times = list(range(4))
+    values = [0.5, 1, 1, 0]
+    ts = TimeSeries.from_lists(times=times, values=values)
+    assert ts.times() == times
+    assert ts.values() == values
+
+
+@pytest.mark.xfail(raises=ValueError)
+def test_from_lists_not_equal_size():
+    times = list(range(4))
+    values = [0.5, 1, 1]
+    TimeSeries.from_lists(times=times, values=values)
+
+
+def test_stitch():
+    times_1 = list(range(4))
+    values_1 = [0.5, 1, 1, 0]
+    time_series_1 = TimeSeries.from_lists(times=times_1, values=values_1)
+
+    times_2 = list(range(4))
+    values_2 = [-0.5, -1, -1, 0]
+    time_series_2 = TimeSeries.from_lists(times=times_2, values=values_2)
+
+    new_ts_mean = time_series_1.stitch(time_series_2, boundary="mean")
+    new_ts_left = time_series_1.stitch(time_series_2, boundary="left")
+    new_ts_right = time_series_1.stitch(time_series_2, boundary="right")
+
+    excepted_times = list(range(7))
+    assert new_ts_mean.times() == excepted_times
+    assert new_ts_left.times() == excepted_times
+    assert new_ts_right.times() == excepted_times
+
+    assert new_ts_mean.values() == [0.5, 1, 1, -0.25, -1, -1, 0]
+    assert new_ts_left.values() == [0.5, 1, 1, 0, -1, -1, 0]
+    assert new_ts_right.values() == [0.5, 1, 1, -0.5, -1, -1, 0]
+
+
+def test_stitch_empty_ts():
```

Comment:
Please test with different lists being zero length (and both being zero length).

For this, and the above tests, I suggest using "@pytest.mark.parametrize"

Reply:
Thanks, included additional tests with either one or both TimeSeries being empty.
I still prefer the inputs for the tests to be defined in each test function, makes it easier to follow the code.